### ISSUE TITLE
[MHV Portal / Accessibility] Print/download menu component with incorrect roles and keyboard support

### ISSUE THREAD

**Issue by sarahhortonadhoc** [department-of-veterans-affairs/va.gov-team#122375]
### 📝 User story
As an assistive technology user, I want interactive components to be provided using the correct roles, so that when I encounter a component I know what it is and know how to operate it.

As a keyboard user, I want interactive components to operate as expected based on their role, so that I can navigate and operate them successfully.

### 📝 Problem statement
- The print/download menu component is provided as buttons rather than as a menu with menu items. 
- Tab and arrow keys move focus through menu items; Tab moves focus out of the component.
- The menu remains open when focus moves out of the component.

### 📝 Details
<details><summary>Links, screenshots, steps to recreate</summary>

##### 🔗 From [Medical Records > Care summaries > ADHC Consult Results](https://staging.va.gov/my-health/medical-records/summaries-and-notes/811)

```
<ul…>
<li><button…>Print this page</button></li>
<li><button…">Download PDF of this page</button></li>
<li><button…>Download a text file (.txt) of this page</button></li>
</ul>
```
#### 👣 Steps to recreate

1. Navigate to the Medical Records > Allergies and reactions page.
2. Open the Print or download menu.
3. Examine the rendered properties of each menu item (use developer tools to view the rendered properties or navigate using a screen reader).
4. Notice that each menu item has the role of button.
5. Navigate through the menu items and menu component using Tab and up/down arrows.
6. Notice that Tab moves through menu items and then to other focusable elements on the page.
7. Notice that up/down arrows move through menu items and then scroll the page.
</details>

---

### 🔗 Point of contact
Sarah Horton (@sarahhortonadhoc)

### ⚠️ Severity level
- 2, Serious. Should be fixed in 1-2 sprints post-launch.

### 📝 Related issues
- #122282
- #122275 

#### Accessibility Standards
- [Guideline 2.1 Keyboard Accessible: Make all functionality available from a keyboard](https://www.w3.org/WAI/WCAG22/Understanding/keyboard-accessible)
  - [SC 2.1.1 Keyboard (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/keyboard)
- [Guideline 2.4 Navigable: Provide ways to help users navigate, find content, and determine where they are](https://www.w3.org/WAI/WCAG21/Understanding/navigable)
  - [SC 2.4.3 Focus Order (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html)
- [Guideline 4.1 Compatible: Maximize compatibility with current and future user agents, including assistive technologies](https://www.w3.org/WAI/WCAG22/Understanding/compatible)
  - [SC 4.1.2 Name, Role, Value (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)

#### Experience Standards
- [Category Usability: User is able to easily know or learn how to use the interface to complete tasks](https://design.va.gov/about/experience-standards/#category-usability)
  - [User can identify an element and its state](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
- [Consistency: User has a predictable and unchanging experience across all VA.gov and VA Mobile App services, tools, and pages](https://design.va.gov/about/experience-standards/#category-consistency)
  - [User experiences predictable interactions with components and patterns](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components-and-patterns)

### 𝌣 Proposed solution or next steps
- Revisit the implementation of the print/download actions menu button to address multiple issues.

### 𝌣 Supporting resources
- [Menu Button Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/) from W3C ARIA Authoring Practices Guide (APG)
- [ARIA: menu role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/menu_role) from MDN
- [Experimental Design [accordion menu print and download (list) button] #1877](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1877) for details about a similar proposed component from Jun 2023

### ✅ Acceptance criteria
- [ ] The system presents the print/download menu component with menu and menu item roles.
- [ ] The system supports standard arrow key navigation through menu items.
- [ ] The system closes the print/download menu component when focus moves out of the component (with Tab or Esc keys).
- [ ] The screen reader user hears the component announced as a menu and menu items announced as menu items.
- [ ] The keyboard user moves focus to and away from the the components using the Tab key.
- [ ] The keyboard user operates the component (opens, closes, and navigates through menu items, selects menu items) using standard keys (arrow keys, spacebar, enter, Esc).

### ✅ Possible tasks
- [ ] Review, refine and acknowledge feedback.
- [ ] Update ticket information (assignee, labels, project(s), etc.)
- [ ] Fix and/or document decisions made.
- [ ] Product manager to move ticket in their respective product board to 'QA/A11y Review' after reviewing documented decisions / validating fix.
- [ ] Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

### 🔎 General resources

- USWDS Docs: [USWDS Documentation](https://designsystem.digital.gov/)
- VADS Docs: [VADS Experience Standards](https://design.va.gov/about/experience-standards/)
- VADS Docs: [VADS Components Documentation](https://design.va.gov/components/)
- VADS Docs: [VADS Components Storybook](https://design.va.gov/storybook/?path=/docs/about-introduction--docs)
- VA Platform Docs: [Accessibility defect severity rubric](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric)
- VA Github Docs: [List of common issue tickets](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/platform/accessibility/common-issues) you can reference, copy, and add to.
- W3C Docs: [Web Content Accessibility Guidelines (WCAG) 2.2](https://www.w3.org/TR/WCAG22/)
- W3C Docs: [WCAG 2.2 Understanding Docs](https://www.w3.org/WAI/WCAG22/Understanding/)

**Comment by steel36** [comment]
@sarahhortonadhoc This appears to be for Medical Records.  Moving it off of Horizon's board.

**Comment by sarahhortonadhoc** [comment]
Work on this ticket is on hold while working through the #125496.

**Comment by sarahhortonadhoc** [comment]
Blocked by blocked exploration ticket: [[MHV Portal / Accessibility] Explore accessible alternatives to print/download actions menu #125496](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125496).
